.circleci/config.yml

```diff
@@ -8,27 +8,34 @@ commands:
         type: string
     steps:
       - checkout
+      - restore_cache:
+          key: -v0-macos-sys-{{ .Branch }}-<< parameters.py_version >>
       - run:
-          name: "Installing Conda"
+          name: "Set up Conda"
           command: |
-            curl -o Miniconda3-latest-MacOSX-x86_64.sh https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
-            bash ./Miniconda3-latest-MacOSX-x86_64.sh -b
+            if [[ -f ~/miniconda3/LICENSE.txt ]] ; then
+              echo "miniconda installed already."
+            else
+              curl -o Miniconda3-latest-MacOSX-x86_64.sh https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
+              bash ./Miniconda3-latest-MacOSX-x86_64.sh -b
+            fi
             ~/miniconda3/bin/conda init bash
       - run:
-          name: "Preparing environment"
+          name: "Brew"
```

Comment:
Can we standardize on:

Resotring environment cache
Preparing environment   (can contain anything we need to do to prepare the env, installing conda, nox, apt-getting etc).
Saving environment cache